### ISSUE TITLE
Epic: Ensure parts of graphical objects essential for understanding content have sufficient contrast - (2036570198)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#7822]
### Violation: 
 Ensure parts of graphical objects essential for understanding content have sufficient contrast across the design system for the following components with the `--calcite-color-border-input` css prop and `border-color-input` tailwind util:
- Form components (`input`) from https://github.com/Esri/calcite-design-system/issues/7847 (2036570491)
    - `checkbox`
    - `combobox`
    - `input`
    - `input-date-picker`
    - `input-number`
    - `input-text`
    - `segmented-control`
    - `select`
    - `text-area`
- `radio-button` (2036570149)
- `rating` (2036570023)
- `slider` uses `--calcite-color-border-2` [light mode ratio of 1.39:1](https://webaim.org/resources/contrastchecker/?fcolor=D4D4D4&bcolor=F8F8F8)
- `switch` (2036570198)
- `stepper item` (2036570233)

This looks to relate to the `--calcite-color-background` [[More details](https://github.com/Esri/calcite-design-system/issues/7847#issuecomment-1781978975)]:
1. `calcite-mode-light` (#F8F8F8) with the outline of the `input` (#949494), where the contrast ratio is [2.85:1](https://webaim.org/resources/contrastchecker/?fcolor=949494&bcolor=F8F8F8).
2. `calcite-mode-dark` (#353535) with the outline of the `input` (#757575), where the contrast ratio is [2.66:1](https://webaim.org/resources/contrastchecker/?fcolor=757575&bcolor=353535)

### calcite-mode-light:
![image](https://github.com/Esri/calcite-design-system/assets/5023024/1d1c588c-b351-4181-b4ae-c455add51ece)
![image](https://github.com/Esri/calcite-design-system/assets/5023024/b5fce549-4c2b-48f1-845f-44aa98cf9d42)

### calcite-mode-dark:
![image](https://github.com/Esri/calcite-design-system/assets/5023024/560118bf-f6e4-43a3-a008-c3040bd0e66d)
![image](https://github.com/Esri/calcite-design-system/assets/5023024/a92622e6-da04-4e7c-a133-588908fce7c5)

### Screenshots
![image](https://github.com/Esri/calcite-design-system/assets/5023024/6e13edda-f43d-4bb2-9146-e156cba84de1)
![image](https://github.com/Esri/calcite-design-system/assets/5023024/1f72cf44-4478-4b3e-8c24-f7badd9765fe)
![image](https://github.com/Esri/calcite-design-system/assets/5023024/a33f5dc4-913c-4adf-a99d-07cfb80f8125)
![image](https://github.com/Esri/calcite-design-system/assets/5023024/c8236678-e1fa-4b4b-b1bc-33551dd2dbf5)
![image](https://github.com/Esri/calcite-design-system/assets/5023024/722921eb-ae80-4bfa-8d05-daa191f02627)

![image](https://github.com/Esri/calcite-design-system/assets/5023024/34d71305-7af4-4469-a6b0-d09f155769ae)
![image](https://github.com/Esri/calcite-design-system/assets/5023024/2ba8ab20-8b7b-440f-8f68-71dd506eb5f9)
![image](https://github.com/Esri/calcite-design-system/assets/5023024/7cac3e82-0c15-4cf9-ada3-3b27eb76ed8a)

### WCAG Reference: 
- [1.4.11 Non-test Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) 

### Severity: 
 5 

### Media Type: 
 Color and Contrast 

--- 
### Areas for Remediation: 
- Title: Switch - Ensure parts of graphical objects essential for understanding content have sufficient contrast - (2036570197) 
- Module: Switch 
**Issue**
There are graphical objects with a color contrast ratio below 3.00: 1.

Foreground: #949494
Background: #F8F8F8
Contrast ratio: 2.86:1

Examples of text with this contrast ratio include:
- Switch images.

**User Impact**
Users with low vision will have difficulty reading this content.

 
**Suggestion**
Ensure parts of graphical objects essential for understanding content have sufficient contrast. The required minimum contrast ratio is 3.00:1.

Common examples of qualifying objects include lines in a chart, meaningful icons, and annotations within an image.

Graphics that require particular presentation to preserve their meaning are exempt from this requirement.

To evaluate color contrast, see the Accessible Color Picker extension: https://www.accessibility.dev/
 
--- 
### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
While the light mode has sufficient contrast of [3.03:1](https://webaim.org/resources/contrastchecker/?fcolor=949494&bcolor=FFFFFF), the dark mode contrast needs some improvement, currently at [2.66:1](https://webaim.org/resources/contrastchecker/?fcolor=757575&bcolor=353535), which _could_ be obtained with [`#808080`](https://webaim.org/resources/contrastchecker/?fcolor=808080&bcolor=353535) and a 3.1:1 ratio for the `--calcite-color-border-input` value, but adding design to this issue for their expertise.

![image](https://github.com/Esri/calcite-design-system/assets/5023024/3122804a-0189-444d-996e-69a0b3f7a5ea)

Prototype: https://codepen.io/geospatialem/pen/abMOvmM

**Comment by geospatialem** [comment]
For the design consideration, please refer to https://github.com/Esri/calcite-design-system/issues/7822#issuecomment-1870658182 cc @SkyeSeitz @ashetland

**Comment by SkyeSeitz** [comment]
This effort will be investigated in the dark mode color considerations of issue #3122 👍🏽
